e2e/tests/e2e_happy_path/HappyPath.spec.ts

```diff
@@ -101,9 +158,135 @@ suite('Ide checks', async () => {
+        await previewWidget.waitPreviewWidgetAbsence();
+    });
+
+    test('Close running terminal processes and tabs', async () => {
+        await terminal.rejectTerminalProcess('run-with-changes');
+        await terminal.closeTerminalTab('run-with-changes');
+
+        await warningDialog.waitAndCloseIfAppear();
+    });
+});
+
+suite('Validation of debug functionality', async () => {
+    test('Open file and activate breakpoint', async () => {
+        await projectTree.expandPathAndOpenFile(pathToJavaFolder, javaFileName);
+        await editor.selectTab(javaFileName);
+        await editor.moveCursorToLineAndChar(javaFileName, 34, 1);
+        await editor.activateBreakpoint(javaFileName, 32);
+    });
+
+    test('Launch debug', async () => {
+        await topMenu.selectOption('Terminal', 'Run Task...');
+        await quickOpenContainer.clickOnContainerItem('che: run-debug');
+
+        await ide.waitNotificationAndConfirm('A new process is now listening on port 8080', 120000);
+        await ide.waitNotificationAndOpenLink('Redirect is now enabled on port 8080', 120000);
+    });
+
+    test('Check content of the launched application', async () => {
+        await previewWidget.waitContentAvailable(SpringAppLocators.springErrorMessageLocator, 60000, 10000);
+    });
+
+    test('Open debug configuration file', async () => {
+        await topMenu.selectOption('Debug', 'Open Configurations');
+        await editor.waitEditorAvailable('launch.json');
+        await editor.selectTab('launch.json');
+    });
+
+    test('Add debug configuration options', async () => {
+        await editor.moveCursorToLineAndChar('launch.json', 5, 22);
+        await editor.performKeyCombination('launch.json', Key.chord(Key.CONTROL, Key.SPACE));
+        await editor.clickOnSuggestion('Java: Launch Program in Current File');
+        await editor.waitTabWithUnsavedStatus('launch.json');
+        await editor.waitText('launch.json', '\"name\": \"Debug (Launch) - Current File\"');
+        await editor.performKeyCombination('launch.json', Key.chord(Key.CONTROL, 's'));
+        await editor.waitTabWithSavedStatus('launch.json');
+    });
+
+    test('Run debug and check application stop in the breakpoint', async () => {
+        await editor.selectTab(javaFileName);
+        await topMenu.selectOption('View', 'Debug');
+        await ide.waitRightToolbarButton(RightToolbarButton.Debug);
+        await debugView.clickOnDebugConfigurationDropDown();
+        await debugView.clickOnDebugConfigurationItem('Debug (Launch) - Current File');
+        await debugView.clickOnRunDebugButton();
+
+        await previewWidget.refreshPage();
+        await editor.waitStoppedDebugBreakpoint(javaFileName, 32);
+    });
 
     test.skip('Github plugin initialization', async () => {
         await githubPlugin.openGitHubPluginContainer();
         await githubPlugin.waitChangesPresence(expectedGithubChanges);
     });
 });
+
+async function checkJavaPathCompletion() {
```

Comment:
I guess this method is supposed to live elsewhere, right?

Reply:
In my opinion no, because this method is too specific for current test